### PR TITLE
hw-2.4

### PR DESCRIPTION
**Pull request by julia-shvabovich**

### CODE REVIEW

File: src/main/java/core/basesyntax/Figure.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    public Figure(int area, Color color) {
+        this.area = area;
+        this.color = color;
+    }
+
+    public abstract void draw();
+
+    public int getArea() {
+        return area;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java]: this method should be custom for each figure, let's make abstract and override it in subclasses

File: src/main/java/core/basesyntax/Figure.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public abstract class Figure {
+    private int area;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java]: we don't need this variable here, area would calculated by method getArea()

File: src/main/java/core/basesyntax/Main.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[(randomNumber(1, 100))];
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Main.java]: 100 and 4 are magic numbers, let's make them constants

File: src/main/java/core/basesyntax/Square.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    public Square(int area, Color color) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Square.java]: let's add field "side" that would be passed from constructor

File: src/main/java/core/basesyntax/Figure.java (hunk 1 above)
Comment (julia-shvabovich) [reply src/main/java/core/basesyntax/Figure.java]: done

File: src/main/java/core/basesyntax/Figure.java (hunk 2 above)
Comment (julia-shvabovich) [reply src/main/java/core/basesyntax/Figure.java]: done

File: src/main/java/core/basesyntax/Main.java (hunk 3 above)
Comment (julia-shvabovich) [reply src/main/java/core/basesyntax/Main.java]: done

File: src/main/java/core/basesyntax/Square.java (hunk 4 above)
Comment (julia-shvabovich) [reply src/main/java/core/basesyntax/Square.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**